### DISCUSSION TITLE
Sleep timer: Auto scheduled, One-click on/off + remember duration, More visual clue when timer is active

### DISCUSSION THREAD

**Discussion by Trikenstein** [Goodwy/PlayBooks#96] [Ideas]
Currently using PlayBook: Audiobook Player v4.0.3. The sleep time is too basic and not really user friendly. The suggestions below are taken from other more advanced players in Google playstore (Smart Audiobook player, Listen Audiobook player, Libby player, Replaio Radio)

1. **Scheduled sleep timer**: need a one time setting with enable/disable flag. If enabled, requires a configurable start/end time. Typically when current time is between the start/end time, the sleep timer is activated automatically. Those who often fall asleep and have the audiobook playing the whole night would certainly appreciate this idea.
2. **One-click on/off + Remember duration**: currently when the timer is selected, a screen pops up to ask for the duration. And another click on the duration to start the timer. It would be more convenient to start the timer immediately when the icon is tapped. The duration is set to the previously chosen duration. The reason is b/c most of the time, people tend to always use the same duration. Still, how could we change the sleep timer duration if it already started on a given duration? Several methods: Smart audiobook player uses a preconfigured duration in the settings. While Listen AB player, when the timer is active, a tap on the icon open a screen asking to cancel or to choose a duration. This seems more flexible but actually presents a minor convenient b/c it requires two taps to stop the timer. You can implement other tricks, double-tap or long press on the icon to open a screen to modify the duration. Single tap is just to on/off the timer.
3. **More visual clue when timer is active**: The idea here is a visual clue to make obvious that the sleep timer is active. Currently the icon just have a bold circle. In other apps where I find the design more intuitive: 
  - A count down counter: the sleep-timer icon is replaced by a count down counter. Ideally displayed in a different font color. Additional advantage: you know the remaining time when the timer will be off. In my opinion, the count down counter is the most intuitive. EDIT (2025-01-05): the countdown counter already exists! Kudos, thanks. But its not noticeable I missed it. All needed now is just to improve its visibility. See my answer below.
  - An icon with an animated "Zzz". This looks fancy. The clue is quite obvious indeed, not sure if it is easy to do.
  - The same icon displayed in an accented color. In your case, the simplest might be to display the current "time active" icon, in a different color.

**Comment by Goodwy** [comment]
Long press the icon to immediately start the sleep timer for a preset time.

**Reply by Trikenstein** [reply]
Hi,

Thanks. Oh cool, and I have now noticed there is a count down timer in overlay on the top right of the cover.  Nevertheless, I think the user experience could be improved:

- Reverse the tap: short-press starts immediately with previously selected duration, long-press opens dialog to select duration. The reason is simply b/c it is more natural to start the sleep timer than to to select duration then start. Especially most of the time people will just confirm the previous duration.
- Make the count down counter on the top right to stand out better. This countdown counter is genius as it is rarely implemented in other apps, thanks. But it can be even better. Maybe bold font with a different color. Yeah that would spoil the visual of the cover a bit. The rationale is that when you activate the sleep timer. You likely will not look at the screen. So spoiling the cover is a moot point. However, if you didn't want the sleep time on and it was already activated, thanks to the clearly visible clue, you WILL know the timer is on.
- Scheduled timer: as requested in the initial post. This is a very convenient feature for Audiobook player.
